test/gradle-test-wrapper/src/test/java/gg/jte/GradleMatrixTest.java

```diff
@@ -40,18 +53,106 @@ public static Stream<Arguments> runGradleBuild() throws IOException {
 
     @ParameterizedTest
     @MethodSource
-    public void runGradleBuild(Path projectDir, String gradleVersion) {
+    public void runGradleBuild(Path projectDir, String gradleVersion) throws IOException {
+        // Clean the configuration cache. Makes it easier to run the tests locally
+        // multiple times in a predictable way. For reference, see:
+        // https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:usage:invalidate
+        deleteDir(projectDir.resolve(".gradle/configuration-cache"));
+
+        // Then run the task
+        BuildResult result = runner(projectDir, gradleVersion, TASK_NAME).build();
+
+        // Check task was successful
+        BuildTask task = result.task(TASK_NAME);
+        Assertions.assertNotNull(task, "Build result must have a task " + TASK_NAME);
+        Assertions.assertNotEquals(
+            TaskOutcome.FAILED,
+            task.getOutcome(),
+            String.format("Build failed in %s with Gradle Version %s", projectDir, gradleVersion)
+        );
+    }
+
+    public static Stream<Arguments> checkBuildCache() {
+        return Stream
+                .of(Paths.get("../jte-runtime-cp-test-models-gradle"), Paths.get("../jte-runtime-test-gradle-convention"))
```

Comment:
Hmm, what about the other test projects?

`gg.jte.GradleMatrixTest#runGradleBuild()`

Reply:
We are only interested in testing that the tasks are cacheable and not checking every project (done in the other above).